### PR TITLE
Introduce and implement container GC policy

### PR DESCRIPTION
**Pull request by vmarmol**

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -120,7 +122,8 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
...
 	fs.BoolVar(&s.EnableDebuggingHandlers, "enable_debugging_handlers", s.EnableDebuggingHandlers, "Enables server endpoints for log collection and local running of containers and commands")
 	fs.DurationVar(&s.MinimumGCAge, "minimum_container_ttl_duration", s.MinimumGCAge, "Minimum age for a finished container before it is garbage collected.  Examples: '300ms', '10s' or '2h45m'")
-	fs.IntVar(&s.MaxContainerCount, "maximum_dead_containers_per_container", s.MaxContainerCount, "Maximum number of old instances of a container to retain per container.  Each container takes up some disk space.  Default: 5.")
+	fs.IntVar(&s.MaxPerPodContainerCount, "maximum_dead_containers_per_container", s.MaxPerPodContainerCount, "Maximum number of old instances of a container to retain per container.  Each container takes up some disk space.  Default: 5.")
+	fs.IntVar(&s.MaxContainerCount, "maximum_dead_containers_per", s.MaxContainerCount, "Maximum number of old instances of a containers to retain globally.  Each container takes up some disk space.  Default: 100.")
```
Comment (dchen1107) [inline cmd/kubelet/app/server.go]: s/maximum_dead_containers_per/maximum_dead_containers_per_node?

**Comment by dchen1107** [comment]
LGTM except small nits. You have to rebase for merge

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (vmarmol) [inline cmd/kubelet/app/server.go]: Sorry I think the "per" was extra. Changed to "maximum_dead_containers" since it is a flag to a particular node.

**Comment by vmarmol** [comment]
Rebased and fixed the nit. Thanks for the review @dchen1107!

**Comment by dchen1107** [comment]
LGTM and will merge on green.

**Comment by vmarmol** [comment]
All is green :)

**Pull request merged**